code/bmpman/bmpman.cpp

```diff
@@ -1663,6 +1773,58 @@ void bm_lock_ani(int handle, int bitmapnum, bitmap_entry *be, bitmap *bmp, ubyte
 	anim_free(the_anim);
 }
 
+
+void bm_lock_apng(int handle, int bitmapnum, bitmap_entry *be, bitmap *bmp, ubyte bpp, ubyte flags) {
+
+	int first_frame = be->info.ani.first_frame;
+	int nframes = bm_bitmaps[first_frame].info.ani.num_frames;
+
+	apng::apng_ani* the_apng;
```

Comment:
Using a `std::unique_ptr` would clean up some code and would make sure that the memory doesn't leak even if an exception is thrown. I don't think there is a way for this code to leak the pointer but using a `unique_ptr` adds a little extra safety.